docs(homeserver): Improve developer docs and readmes (#431)
pubky/pubky-core: README.md

<h1 align="center"><a href="https://pubky.org/"><img alt="Pubky" src="./.svg/pubky-core-logo.svg" width="200" /></a></h1>

<h3 align="center">
  Homeserver and SDKs for Pubky.
</h3>

<div align="center">
  <h3>

>     <a href="https://docs.pubky.org/">Docs</a>

Author: ```suggestion
    <a href="https://pubky.org/">Docs</a>
```

    <span> | </span>
    <a href="https://docs.rs/pubky">Rust SDK</a>
    <span> | </span>
    <a href="https://www.npmjs.com/package/@synonymdev/pubky">JavaScript SDK</a>
  </h3>
  <a href="https://github.com/pubky/pubky-core/blob/main/LICENSE"><img src="https://img.shields.io/github/license/pubky/pubky-core" alt="GitHub License" /></a>
  <a href="https://github.com/pubky/pubky-core/releases/latest/"><img src="https://img.shields.io/github/v/release/pubky/pubky-core" alt="GitHub Release" /></a>
  <a href="https://crates.io/crates/pubky"><img src="https://img.shields.io/crates/v/pubky" alt="Crates.io Version" /></a>
  <a href="https://www.npmjs.com/package/@synonymdev/pubky"><img src="https://img.shields.io/npm/v/@synonymdev/pubky" alt="npm Version" /></a>
  <br/>
  <a href="https://t.me/pubkycore"><img src="https://img.shields.io/badge/Chat-Telegram-violet" alt="Telegram Chat Group" /></a>
  <a href="https://deepwiki.com/pubky/pubky-core"><img src="https://deepwiki.com/badge.svg" alt="Ask DeepWiki" /></a>
</div>

> [Pubky](https://pubky.org) is an open protocol for building censorship-resistant applications where users own their identity, data, and connections. No platform lock-in, no algorithmic feeds you don't control, no losing everything when a service shuts down. Your keys are your identity, and you choose where your data lives.

Author: Can we get more space betweet the badges and this text? It's too sticky currently.

<img width="1892" height="924" alt="Image" src="https://github.com/user-attachments/assets/4b8cbd4b-42ff-4582-8d92-44d778b38a08" />

> [Pubky](https://pubky.org) is an open protocol for building censorship-resistant applications where users own their identity, data, and connections. No platform lock-in, no algorithmic feeds you don't control, no losing everything when a service shuts down. Your keys are your identity, and you choose where your data lives.

Author: Do I control the algorithmic feed in Pubky.app? 

```suggestion
[Pubky](https://pubky.org) is an open protocol for building censorship-resistant applications where users own their identity, data, and connections. No platform lock-in, no losing everything when a service shuts down. Your keys are your identity, and you choose where your data lives.
```

To learn more about the vision, see [What is Pubky?](https://pubky.org/tldr/), [Censorship Resistance](https://pubky.org/explore/concepts/censorship/), and [Credible Exit](https://pubky.org/explore/concepts/credible-exit/).

This repository contains the core infrastructure: a homeserver that stores and serves user data, Rust and JavaScript SDKs for building apps, a local testnet, and examples.

## Who Is This For?

- **App developers**: Use the [SDK](https://pubky.org/explore/pubkycore/sdk/) to build apps that read and write user data on homeservers. Start with [Local Development](./docs/LOCAL_DEVELOPMENT.md) and the [examples](./examples).
- **Operators**: [Install and run a homeserver](./docs/INSTALL.md) to host user data.

> - **Contributors** [Set up a local testnet](./docs/LOCAL_DEVELOPMENT.md) and [run the tests](./docs/TESTING.md) to develop on pubky-core itself.

Author: Please again, do not mention contributors. We really don't want to encourage them. They are more pain than good.

## What Is a Homeserver?

> A Pubky homeserver stores and serves user data for Pubky app. Users choose which homeserver holds their data, and can move to another at any time. The homeserver exposes HTTP APIs for authenticated writes and public reads, and publishes [PKARR](https://github.com/pubky/pkarr) records so other clients can discover where a user's data lives.

Author: "Pubky app": Do you mean pubky.app or any Pubky application?

> - Public-key based signup, signin and third-party app authorization

Author: ```suggestion
- Public-key-based signup, sign-in, and third-party app authorization.
```

<h1 align="center"><a href="https://pubky.org/"><img alt="Pubky" src="./.svg/pubky-core-logo.svg" width="200" /></a></h1>

<h3 align="center">

>   Reference Homeserver implementation for Pubky.

Author: I am not sure we should call it "Reference". With "Reference" it feels like we want to encourage alternative implementations and I am not sure we should do this already. At the moment, we can just be reckless and dirty with changes. If we have other implementations that use us as reference, this gets more difficult.

So I think we should remove "Reference" here for at least a while. We can add it back when we are more stable IMO.

</h3>

<div align="center">
  <h3>
    <a href="https://docs.pubky.org/">Docs</a>
    <span> | </span>
    <a href="https://docs.rs/pubky">Rust SDK</a>
    <span> | </span>
    <a href="https://www.npmjs.com/package/@synonymdev/pubky">JavaScript SDK</a>
  </h3>
  <a href="https://github.com/pubky/pubky-core/blob/main/LICENSE"><img src="https://img.shields.io/github/license/pubky/pubky-core" alt="GitHub License" /></a>
  <a href="https://github.com/pubky/pubky-core/releases/latest/"><img src="https://img.shields.io/github/v/release/pubky/pubky-core" alt="GitHub Release" /></a>
  <a href="https://crates.io/crates/pubky"><img src="https://img.shields.io/crates/v/pubky" alt="Crates.io Version" /></a>
  <a href="https://www.npmjs.com/package/@synonymdev/pubky"><img src="https://img.shields.io/npm/v/@synonymdev/pubky" alt="npm Version" /></a>
  <br/>
  <a href="https://t.me/pubkycore"><img src="https://img.shields.io/badge/Chat-Telegram-violet" alt="Telegram Chat Group" /></a>
  <a href="https://deepwiki.com/pubky/pubky-core"><img src="https://deepwiki.com/badge.svg" alt="Ask DeepWiki" /></a>
</div>

> ## What Is This Repository?

Author: Unnecessary title?

```suggestion
```

This repository contains the reference Pubky homeserver implementation and the crates needed to run, test, and integrate with it - including the Rust and JavaScript SDKs, a local testnet, and examples.

## What Is a Homeserver?

A Pubky Homeserver stores and serves user data. It exposes HTTP APIs for authenticated writes and public reads, and publishes DNS records so other clients can discover where a user's data lives.

- Public-key based signup and signin
- Third-party authorization through Pubky auth flows
- File storage via HTTP `PUT`, `GET`, `DELETE`, and listing APIs (WebDAV-like)

> - PKDNS/Pkarr publishing for homeserver discovery

Author: Always write PKARR with capital letters. Please check the rest of the PR on this too.

```suggestion
- PKDNS/PKARR publishing for homeserver discovery
```

- Admin and metrics endpoints for operators

## Repository Layout

| Path | Purpose |
| --- | --- |
| [`pubky-homeserver`](./pubky-homeserver) | Homeserver binary and library crate. |
| [`pubky-sdk`](./pubky-sdk) | Rust client for Pubky apps, plus JS/WASM bindings. |
| [`pubky-common`](./pubky-common) | Shared types and helpers used by the SDK and homeserver. |
| [`pubky-testnet`](./pubky-testnet) | Local ephemeral Pubky network for development and tests. |
| [`examples`](./examples) | Rust and JavaScript examples for signup, auth, storage, and requests. |
| [`e2e`](./e2e) | End-to-end tests covering cross-crate workflows. |
| [`docs`](./docs) | Install guides, local development, and testing docs. |

## Getting Started

| I want to... | Guide |
| --- | --- |
| Run a homeserver | [Install and Run Pubky Homeserver](./docs/INSTALL.md) |
| Develop locally against a testnet | [Local Development](./docs/LOCAL_DEVELOPMENT.md) |

> | Run tests and CI | [Testing](./docs/TESTING.md) |

Author: This two columns feel redundant. `Run a homeserver` is very simillar to `Install and run Pubky Homeserver`. Should we turn this into a simple list?

```suggestion
- [Install and run a homeserver](./docs/INSTALL.md) |
- [Develop locally against a testnet](./docs/LOCAL_DEVELOPMENT.md) |
- [Run tests and CI](./docs/TESTING.md) |
```

## Use the SDK

The SDK is the easiest way to interact with a homeserver from your app.

Rust:

```toml
[dependencies]
pubky = "0.9"  # see https://crates.io/crates/pubky for latest
```

JavaScript and TypeScript:

```bash
npm install @synonymdev/pubky
```

See the [`pubky-sdk` README](./pubky-sdk) for API details or browse the [examples](./examples).

## Links

- [Pubky website](https://pubky.org/)

> - [Documentation](https://docs.pubky.org/)

Author: docs.pubky.org is now simply pubky.org.

> - [Pkarr](https://pkarr.org/)

Author: This is Ars domain. Please link to our Git repo https://github.com/pubky/pkarr

> - [Contributors Guide](./CONTRIBUTORS.md)

Author: I would not draw too much attention on the Contributors guide. External contributors are more pain than good in my experience. Can we remove this line?

- [Release Process](./RELEASING.md)

---

> May the power ⚡ be with you. Powered by [pkarr](https://github.com/pubky/pkarr).

Author: PKARR mentions are getting a bit excessive here IMO
